packages/babel-plugin-minify-builtins/src/index.js

```diff
@@ -147,28 +147,43 @@ function getSegmentedSubPaths(paths) {
   paths[0].getDeepestCommonAncestorFrom(
     paths,
     (lastCommon, index, ancestries) => {
-      // we found the LCA
-      if (!lastCommon.isProgram()) {
-        lastCommon = !lastCommon.isFunction()
-          ? lastCommon.getFunctionParent()
-          : lastCommon;
+      // found the LCA
+      if (!lastCommon.isProgram() && lastCommon.isFunction()) {
         segments.set(lastCommon, paths);
         return;
       }
+      // Found the LCA in the parent
+      let funParent = lastCommon.getFunctionParent();
+      if (!(lastCommon.isProgram() || funParent.isProgram())) {
+        segments.set(funParent, paths);
+        return;
+      }
```

Comment:
in both the above conditions, `lastCommon.isProgram()` should be false. So, why not combine this logic

```js
let fnParent = lastCommon.getFunctionParent();
if (
  !lastCommon.isProgram() &&
  (lastCommon.isFunction() || !lastCommon.getFunctionParent().isProgram())
) {
  segments.set(lastCommon, paths);
  return;
}
```